### PR TITLE
add example of user-data with private network

### PR DESCRIPTION
**Pull request by tarcisoferreira**
Example of user-data with private network

### CODE REVIEW

File: user-data/private-network/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+### Configure Private Networks
+
+This example demonstrates how to set up a private network using user data, requiring the input of the VID and subnet address.
```
Comment (esoubihe) [inline user-data/private-network/README.md:3]:
```suggestion
This example demonstrates how to add a server to a previously configured [private network](https://www.latitude.sh/docs/networking/private-networks) as soon as it boots for the first time.

This is helpful for automating the process of adding servers to a private network without the need for a custom setup.

### Requirements:
* Create a VLAN by going to Networking -> Private networks. Write down the VID value.
* Replace the `vid` variable in the script with the VID of the VLAN to which you want the server to be added.
* Replace the `subnet_address` variable with a private IP for internal communication.
```

File: user-data/private-network/data.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#cloud-config
+packages:
+  - snapd
+
+runcmd:
+  - snap install yq
+  - |
+    vid=10
```
Comment (esoubihe) [inline user-data/private-network/data.yaml:8]:
```suggestion
    vid=<VID> # replace with the VID of the vlan you created
```

File: user-data/private-network/data.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    subnet_address="10.0.0.2/24"
```
Comment (esoubihe) [inline user-data/private-network/data.yaml:9]:
```suggestion
    subnet_address="<IP_ADDRESS>" # replace with a private IP address followed by the subnet, e.g. 10.0.0.2/24
```

File: user-data/private-network/data.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    iface=$(cat /etc/netplan/50-cloud-init.yaml | yq e '.network.ethernets[] | select(has("gateway4") | not ) | path | .[-1]' -)
+    cat /etc/netplan/50-cloud-init.yaml | yq e ".network.vlans.vlan10 = {\"id\": \"$vid\", \"link\": \"$iface\", \"addresses\": [\"$subnet_address\"]}"  >> /etc/netplan/network-config.yaml
```
Comment (esoubihe) [inline user-data/private-network/data.yaml:11]: I'm guessing we should replace `vlan10` with `vlan<VID>`

**Review by tarcisoferreira — COMMENTED** [review]
Modificações aprovadas.

File: user-data/private-network/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+### Configure Private Networks
+
+This example demonstrates how to add a server to a previously configured [private network](https://www.latitude.sh/docs/networking/private-networks) as soon as it boots for the first time.
+
+This is helpful for automating the process of adding servers to a private network without the need for a custom setup.
+
+### Requirements:
+* Create a VLAN by going to Networking -> Private networks. Write down the VID value.
+* Replace the `vid` variable in the script with the VID of the VLAN to which you want the server to be added.
```
Comment (esoubihe) [inline user-data/private-network/README.md:9]:
```suggestion
* Replace the `<VID>` variable in the script with the VID of the VLAN to which you want the server to be added.
```

File: user-data/private-network/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+* Replace the `subnet_address` variable with a private IP for internal communication.
```
Comment (esoubihe) [inline user-data/private-network/README.md:10]:
```suggestion
* Replace `<IP_ADDRESS>` variable with a private IP for internal communication.
```

**Review by esoubihe — APPROVED** [review]

**Pull request merged**